### DISCUSSION TITLE
Purpose of this package over Microsoft.AspNetCore.Components.WebAssembly.Authentication

### DISCUSSION THREAD

**Discussion by StefanOverHaevgRZ** [WildGums/Blorc.OpenIdConnect#607] [Q&A] [answered]
Hello there,

I was redirected here from an IdP documentation page, saying one should use the `Blorc.OpenIdConnect` package over `Microsoft.AspNetCore.Components.WebAssembly.Authentication`. Neither that site, nor the readme of this repository provide any clue to the **why**.

The readme simply states:
> The right way to use OpenIdConnect on Blazor.

Could you show up the issues with the existing package provided by Microsoft, and how this package does things better? 🤓 

With kind regards
Stefan

**Comment by alexfdezsauco** [comment, marked as answer]
1) Who told you that you 'must' use this package instead of `Microsoft.AspNetCore.Components.WebAssembly.Authentication`? Whoever told you might know something 😉

2) The statement `The right way to use OpenIdConnect on Blazor` is just a slogan, but to discover "why" I invite you to run the example application, and explore the library features 

3) We implemented this because we had previous experience using `oidc-client.js` from https://github.com/IdentityModel/oidc-client-js we keep up to date with its successor https://github.com/authts/oidc-client-ts. From our experience, `oidc-client.js` is a great implementation of OpenIDConnect. Probably the implementation in `AuthenticationService.js` from Microsoft.AspNetCore.Components.WebAssembly.Authentication is also a correct one. 

4) If you want to use `Microsoft.AspNetCore.Components.WebAssembly.Authentication` instead of `Blorc.OpenIdConnect`, go ahead… but keep in mind that you could miss something 😉. Seriously, both options are valid, and the choice depends on your specific needs. Good luck with your decision!

**Reply by StefanOverHaevgRZ** [reply]
Hey @alexfdezsauco, thanks for the reply! 🙂

1.  e.g. https://docs.logto.io/sdk/dotnet-core/blazor-wasm/ suggests oone should use this package, especially pointing out not to use the Microsoft package. I'll hit them up regarding their ideas, too :)

2. Okay, thanks, I'll look at that!
